docs/FormatsAndConventions.md

```diff
@@ -138,6 +138,48 @@ e.g. `Error alert email has very long subject #5958`
 > to compensate for GitHub's lack of strong linking between the two.
 > Assume there is an invisible prefix in front of the PR title `Fixes issue : ...`
 
+## References to code elements
+
+Follow these conventions when referring to code elements **from a non-code context** e.g. when referring to a function 
+name from a commit message.
+
+> The objective is to be as concise as possible without causing ambiguities. Therefore, omit optional details when
+> those details are not pertinent to the context.
+
+<p>
+
+> Refer to the respective coding standards for conventions on how to refer to code elements _from code contexts_ 
+> e.g. when referring to a function from a code comment. 
+
+### Java
+
+* Variables: `package.class#variable` (optional: `package`)
+
+  > Examples:
+  > 
+  > * `seedu.address.data.Person#name`
+  > * `Person#name`  -- optional parts omitted
+
+* Methods: `package.class#method(paramTypes):returnType` (optional: `package`, `:returnType`)
```

Comment:
Will remove `:` as the examples confirm that we don't include the separator symbol if the component is omitted.